### PR TITLE
[DOC] updates documentation related to java compatibility [skip ci]

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR updates docs/download.md regarding Java compatibility.

signed-off-by: Suraj Aralihalli <suraj.ara16@gmail.com>

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -31,19 +31,22 @@ The plugin is tested on the following architectures:
 
 	NVIDIA Driver*: 470+
 
-	Python 3.6+, Scala 2.12, Java 8, Java 11, Java 17
+	Runtime: 
+		Scala 2.12
+		Python, Java Virtual Machine (JVM) compatible with your spark-version. 
+
+		* Check the Spark documentation for Python and Java version compatibility at `https://spark.apache.org/docs/{spark-version}/`. Please be aware that we do not currently support Spark builds with Scala 2.13.
```
Comment (gerashegalov) [inline docs/download.md:38]: We strive to keep the line length under around 100 characters. 

Optionally could add  a link to a concrete Spark version: like the latest or to the earliest

**Review by gerashegalov — DISMISSED** [review]
LGTM, some nits

File: docs/download.md (hunk 1 above)
Comment (viadea) [reply docs/download.md:38]: yes let's make it at most 100 characters here.

**Review by gerashegalov — DISMISSED** [review]

File: docs/download.md (hunk 1 above)
Comment (SurajAralihalli) [reply docs/download.md]: My bad, I have updated the PR

**Review by gerashegalov — APPROVED** [review]

File: docs/download.md [hunk 2]
```diff
@@ -31,19 +31,24 @@ The plugin is tested on the following architectures:
...
 	NVIDIA Driver*: 470+
 
-	Python 3.6+, Scala 2.12, Java 8, Java 11, Java 17
+	Runtime: 
+		Scala 2.12
+		Python, Java Virtual Machine (JVM) compatible with your spark-version. 
+
+		* Check the Spark documentation for Python and Java version compatibility with your specific 
+		Spark version. For instance, visit `https://spark.apache.org/docs/3.4.1` for Spark 3.4.1. 
+		Please be aware that we do not currently support Spark builds with Scala 2.13.
```
Comment (sameerz) [inline docs/download.md:40]: Minor nit, we already specify we do not support Scala 2.12, I do not think we need to call out what we do not support.  That is, I think this line can be removed.

Comment (gerashegalov) [reply docs/download.md:40]: The idea is was to highlight that the plugin does not have special requirements for anything but Scala.

**Review by sameerz — APPROVED** [review]

**Pull request merged**